sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,228 @@
+                                                    "before transforming data."
+        discretized = list()
+        for index, col in enumerate(X.T):
+            if index not in self.continuous_columns:
+                discretized.append(col.reshape(len(X), 1))
+            transformed = self.cts2cat(index, col)
+            discretized.append(transformed.reshape(len(X), 1))
+        return np.hstack(discretized)
+
+    def fit_transform(self, X, Y):
+        """Performs the fitting and the data transformations.
+        """
+        self.fit(X, Y)
+        return self.transform(X)
+
+    def cat2intervals(self, index, col):
+        """Converts a categorical column into an interval. This is the naive
+        linear search method, but in practice, because there won't be many
+        intervals to search through (usually at most 5), this works effectively
+        without an interval tree.
+        """
+        mappings = self.intervals_[index]
+        out_intervals = list()
+        for cat in col:
+            for interval, currcat in mappings:
+                if cat == currcat:
+                    out_intervals.append(interval)
+                    break
+        return out_intervals
+
+    def bin(self, index, attr):
+        """Bins an attribute into its appropriate interval.
+        """
+        mappings = self.intervals_[index]
+        for (a, b), _ in mappings:
+            if a < attr <= b: return a, b
+        raise ValueError("Numeric value did not fit in any interval")
+
+    def cts2cat(self, index, col):
+        """Converts each continuous value into a categorical value.
+        """
+        mappings = self.intervals_[index]
+        categorical = list()
+        for attr in col:
+            for (a, b), cat in mappings:
+                if a < attr <= b:
+                    categorical.append(cat)
+                    break
+        assert len(categorical) == len(col)
+        return np.array(categorical)
+
+    def _mdlp(self, attributes, Y):
+        """
+        *attributes*: A numpy 1 dimensional ndarray
+
+        *Y*: A python list of numeric class labels.
+
+        Returns a list of intervals, based upon the MDLP stopping criterion.
+        """
+
+        order = np.argsort(attributes)
+        x = attributes[order]
+        y = Y[order]
+
+        intervals = list()
+
+        def get_cut(ind):
+            return (x[ind-1] + x[ind]) / 2
+
+        # Now we do a depth first search to create intervals
+        num_samples = len(x)
+        search_intervals = list()
+        search_intervals.append((0, num_samples, 0))
+        while len(search_intervals) > 0:
+            start, end, depth = search_intervals.pop()
+
+            k = self._find_cut(y, start, end)
+
+            # Need to see whether the "front" and "back" of the intervals need
+            # to be float("-inf") or float("inf")
+            if (k == -1) or (self._reject_split(y, start, end, k) and
```

Comment:
Swapping `depth >= self.min_depth` and `_reject_split` is a good suggestion. 

I think that we should keep the name `_reject_split` though. `_validate_split`, semantically, seems to check whether an input is correct, whereas `_reject_split` semantically means "we should stop cutting".

Do you have any follow up comments?